include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -130,6 +130,27 @@ NBL_CONSTEXPR_INLINE_FUNC typename unsigned_integer_of_size<sizeof(T)>::type ext
 	return ieee754::impl::bitCastToUintType(x) & traits<AsFloat>::signMask;
 }
 
+template <typename FloatingPoint NBL_FUNC_REQUIRES(hlsl::is_floating_point_v<FloatingPoint> && hlsl::is_scalar_v<FloatingPoint>)
+NBL_CONSTEXPR_INLINE_FUNC FloatingPoint copySign(FloatingPoint to, FloatingPoint from)
+{
+	using AsUint = typename unsigned_integer_of_size<sizeof(FloatingPoint)>::type;
+
+	const AsUint toAsUint = ieee754::impl::bitCastToUintType(to);
+	const AsUint fromAsUint = ieee754::impl::bitCastToUintType(from);
+
+	return bit_cast<FloatingPoint>(toAsUint | extractSignPreserveBitPattern(from));
```

Comment:
this does not copy sign, because you need to clear the sign bit on `toAsUint` first

right now it will make `to` negative if any of the inputs are negative

btw are you sure you want to force that `FloatingPoint` is non-emulated and scalar? It could work on any float even emulated 64

Reply:
done here: https://github.com/Devsh-Graphics-Programming/Nabla/pull/817